app/src/main/java/com/odoo/orm/sync/providers/ContactProvider.java

```diff
@@ -2,42 +2,167 @@
+                long new_id = db.insert(oModel.getTableName(), null, values);
+                newUri = Uri.withAppendedPath(uri, new_id + "");
+                break;
+            case SINGLE_ROW:
+                throw new UnsupportedOperationException(
+                        "Insert not supported on URI: " + uri);
+            case UriMatcher.NO_MATCH:
+                break;
+            default:
+                throw new UnsupportedOperationException("Unknown uri: " + uri);
+        }
+        assert newUri != null;
+        notifyDataChange(newUri);
+        return newUri;
     }
 
     @Override
     public int delete(Uri uri, String selection, String[] selectionArgs) {
-        return 0;
+        createDBObject(uri);
+        setMatcher(uri);
+        int match = matcher.match(uri);
+        int count = 0;
+        switch (match) {
+            case COLLECTION:
+                SQLiteDatabase db = oModel.getWritableDatabase();
+                count = db.delete(oModel.getTableName(), selection, selectionArgs);
+                break;
+            case SINGLE_ROW:
+                db = oModel.getWritableDatabase();
+                String row_id = uri.getLastPathSegment();
+                count = db.delete(oModel.getTableName(), "_id = ?",
+                        new String[]{row_id});
+                break;
+            case UriMatcher.NO_MATCH:
+                break;
+            default:
+                throw new UnsupportedOperationException("Unknown uri: " + uri);
+        }
+        notifyDataChange(uri);
+        return count;
     }
 
     @Override
-    public int update(Uri uri, ContentValues values, String selection, String[] selectionArgs) {
-        return 0;
+    public int update(Uri uri, ContentValues values, String selection,
+                      String[] selectionArgs) {
+        createDBObject(uri);
+        setMatcher(uri);
+        int count = 0;
+        int match = matcher.match(uri);
+        switch (match) {
+            case COLLECTION:
+                SQLiteDatabase db = oModel.getWritableDatabase();
+                count = db.update(oModel.getTableName(), values, selection,
+                        selectionArgs);
+                break;
+            case SINGLE_ROW:
+                db = oModel.getWritableDatabase();
+                String row_id = uri.getLastPathSegment();
+                count = db.update(oModel.getTableName(), values, "_id = ?",
+                        new String[]{row_id});
+                break;
+            case UriMatcher.NO_MATCH:
+                break;
+            default:
+                throw new UnsupportedOperationException("Unknown uri: " + uri);
+        }
+        notifyDataChange(uri);
+        return count;
+    }
+
+    private void notifyDataChange(Uri uri) {
+        Context ctx = getContext();
+        assert ctx != null;
+        ctx.getContentResolver().notifyChange(uri, null);
     }
 }
```

Comment:
I think this provider can be simplified.
App is for contacts (ResPartner) only. So does not required to build separate authority and URI. We have only one model. So directly create object of that model. and remove extra uri buildup and checking.